File: app/src/main/java/dev/ragnarok/fenrir/push/NotificationScheduler.java
package dev.ragnarok.fenrir.push;

import java.util.concurrent.Executors;

import dev.ragnarok.fenrir.Injection;
import io.reactivex.rxjava3.core.Scheduler;
import io.reactivex.rxjava3.core.SingleTransformer;
import io.reactivex.rxjava3.schedulers.Schedulers;

public class NotificationScheduler {

    public static final Scheduler INSTANCE = Schedulers.from(Executors.newFixedThreadPool(1));

    public static <T> SingleTransformer<T, T> fromNotificationThreadToMain() {
        return single -> single
                .subscribeOn(INSTANCE)
                .observeOn(Injection.provideMainThreadScheduler());
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/push/NotificationScheduler.kt
package dev.ragnarok.fenrir.push

import dev.ragnarok.fenrir.Includes.provideMainThreadScheduler
import io.reactivex.rxjava3.core.Scheduler
import io.reactivex.rxjava3.core.Single
import io.reactivex.rxjava3.core.SingleTransformer
import io.reactivex.rxjava3.schedulers.Schedulers
import java.util.concurrent.Executors

object NotificationScheduler {
    val INSTANCE: Scheduler = Schedulers.from(Executors.newFixedThreadPool(1))
    fun <T : Any> fromNotificationThreadToMain(): SingleTransformer<T, T> {
        return SingleTransformer { single: Single<T> ->
            single
                .subscribeOn(INSTANCE)
                .observeOn(provideMainThreadScheduler())
        }
    }
}